### ISSUE TITLE
docker_container state not functioning as expected

### ISSUE THREAD

**Issue by ntwrkguru** [ansible/ansible-modules-core#4027]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = 
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

Source: Mac OS X 10.11.5 and Ansible via brew
Target: Ubuntu 14.04 on AWS
##### SUMMARY

In a task, when using docker_container module and specifying state, the observed results are different from the expected results. Specifically, state: started I expect to not restart a running container when the configuration of said container has not changed.
##### STEPS TO REPRODUCE

```
  - name: Start Gitlab as a container
    docker_container:
      name: gitlab
      image: gitlab/gitlab-ee:latest
      exposed_ports:
        - 80
        - 443
        - 22
      published_ports:
        - 443:443
        - 80:80
        - 22:22
      hostname: gitlab.example.com
      volumes:
        - /opt/gitlab/config:/etc/gitlab
        - /opt/gitlab/logs:/var/log/gitlab
        - /opt/gitlab/data:/var/opt/gitlab
        - /opt/gitlab/certs:/etc/gitlab/ssl
      state: started
      restart_policy: always
    notify:
      - restart gitlab

  handlers:
  - name: restart gitlab
    shell: "docker exec -i gitlab bash -l -c '/opt/gitlab/bin/gitlab-ctl reconfigure' "
```

##### EXPECTED RESULTS

I expect that I should be able to run this task and that the container would not be restarted if there are no changes to it.
##### ACTUAL RESULTS

```
TASK [Start Gitlab as a container] *********************************************
changed: [gitlab-dev.example.com]

RUNNING HANDLER [restart gitlab] ***********************************************
changed: [gitlab-dev.example.com]

PLAY RECAP *********************************************************************
gitlab-dev.aslab.juniper.net : ok=6    changed=2    unreachable=0    failed=0   
```

**Comment by cpatry-poly** [comment]
Same issue here !

When I run this task :

```
- name: Pull and run cadvisor container
  docker_container:
    name: cadvisor
    image: "google/cadvisor:v0.23.2"
    restart: no
    restart_policy: always
    state: started
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /var/lib/docker/:/var/lib/docker:ro
    ports:
      - "18080:8080"
```

The docker is ALWAYS restarted.

I found that if I remove the 

```
restart_policy: always
```

The docker isn't restarted anymore, but I need this policy !

```
ansible --version
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

**Comment by ntwrkguru** [comment]
Also just found that restart_policy: unless-stopped is invalid, even though it is documented as being an option. [edit: filed issue #4028]

```
fatal: [gitlab-dev.aslab.juniper.net]: FAILED! => {"changed": false, "failed": true, "msg": "value of restart_policy must be one of: on-failure,always, got: unless-stopped"}
```

| parameter | required | default | choices | comments |
| --- | --- | --- | --- | --- |
| restart_policy | no | on-failure | always | Container restart policy. Place quotes around no option. |
|  |  | False |  |  |
|  |  | on-failure |  |  |
|  |  | unless-stopped |  |  |

**Comment by ntwrkguru** [comment]
Looks like this is fixed in pull #3785. Mea culpa.
